### PR TITLE
feat(osctl): add ability to create docker based clusters

### PR DESCRIPTION
**Pull request by andrewrynhard**
Signed-off-by: Andrew Rynhard <andrew@andrewrynhard.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -158,6 +158,10 @@ talos: buildkitd
 		$(COMMON_ARGS)
 	@docker load < build/$@.tar
 
+.PHONY: integration
+integration:
```
Comment (andrewrynhard) [inline Makefile]: This PR moves the integration test to using `osctl cluster create`.

File: hack/dev/userdata/.master-1.tpl.yaml [hunk 2]
```diff
@@ -31,20 +31,8 @@ services:
       ---
       apiVersion: kubelet.config.k8s.io/v1beta1
       kind: KubeletConfiguration
-      failSwapOn: false  # necessary for some docker hosts
```
Comment (andrewrynhard) [inline hack/dev/userdata/.master-1.tpl.yaml]: These are set in `init` when in a container now.

File: hack/dev/userdata/.master-2.tpl.yaml [hunk 3]
```diff
@@ -27,10 +27,6 @@ services:
       nodeRegistration:
         name: master-2
         criSocket: /run/containerd/containerd.sock
-    ignorePreflightErrors:
```
Comment (andrewrynhard) [inline hack/dev/userdata/.master-2.tpl.yaml]: These are set in `init` when in a container now.

File: hack/dev/userdata/.master-3.tpl.yaml [hunk 4]
```diff
@@ -27,10 +27,6 @@ services:
       nodeRegistration:
         name: master-3
         criSocket: /run/containerd/containerd.sock
-    ignorePreflightErrors:
```
Comment (andrewrynhard) [inline hack/dev/userdata/.master-3.tpl.yaml]: These are set in `init` when in a container now.

File: hack/dev/userdata/.worker.tpl.yaml [hunk 5]
```diff
@@ -23,10 +23,6 @@ services:
         criSocket: /run/containerd/containerd.sock
         kubeletExtraArgs:
           node-labels: node-role.kubernetes.io/worker=
-    ignorePreflightErrors:
```
Comment (andrewrynhard) [inline hack/dev/userdata/.worker.tpl.yaml]: These are set in `init` when in a container now.

File: .drone.yml [hunk 6]
```diff
@@ -8,6 +8,8 @@ services:
   volumes:
   - name: dockersock
     path: /var/run
+  - name: tmp
```
Comment (andrewrynhard) [inline .drone.yml]: Required by integration test.

File: cmd/osctl/cmd/cluster/pkg/node/node.go [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+		EndpointsConfig: map[string]*network.EndpointSettings{
+			clusterName: {
+				NetworkID: clusterName,
+			},
+		},
+	}
+
+	// Mutate the container configurations based on the node type.
+
+	switch req.Type {
```
Comment (bradbeam) [inline cmd/osctl/cmd/cluster/pkg/node/node.go]: super nit, but unsure if seeing a switch/fallthrough is cleaner/clearer than if x && y

File: go.sum [hunk 8]
```diff
@@ -30,6 +30,10 @@ github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible h1:dvc1KSkIYTVjZgHf/CTC2diTYC8PzhaA5sFISRfNVrE=
 github.com/docker/distribution v2.7.1-0.20190205005809-0d3efadf0154+incompatible/go.mod h1:J2gT2udsDAN96Uj4KfcMRqY0/ypR+oyYUYmja8H+y+w=
+github.com/docker/docker v1.13.1 h1:IkZjBSIc8hBjLpqeAbeE5mca5mNgeatLHBy3GO78BWo=
+github.com/docker/docker v1.13.1/go.mod h1:eEKB0N0r5NX/I1kEveEz05bcu8tLC/8azJZsviup8Sk=
```
Comment (bradbeam) [inline go.sum]: This seems like a super old version of the docker client library?  Should it be moby/moby ?

Comment (andrewrynhard) [reply go.sum]: The docker deps are super weird. First of all they seemed to stop tagging at 17.03. Then when trying to use 17.03 it isn't picked up as a tag. As far as I could tell, people tend to use v1.13.1 often. I really don't know what we should do here.

File: cmd/osctl/cmd/cluster/pkg/node/node.go (hunk 7 above)
Comment (andrewrynhard) [reply cmd/osctl/cmd/cluster/pkg/node/node.go]: The switch statement is idiomatic go: https://golang.org/doc/effective_go.html#if.

File: go.sum (hunk 8 above)
Comment (andrewrynhard) [reply go.sum]: I definitely think we should follow up on this and figure out how docker expects us to use the latest packages. `moby/moby` straight up didn't compile for me.

**Pull request merged**